mypy/constraints.py

```diff
@@ -444,7 +444,8 @@ def infer_constraints_from_protocol_members(self, res: List[Constraint],
         for member in protocol.type.protocol_members:
             inst = mypy.subtypes.find_member(member, instance, subtype)
             temp = mypy.subtypes.find_member(member, template, subtype)
-            assert inst is not None and temp is not None
+            if inst is None or temp is None:
```

Comment:
I'm not familiar with this code. Do you know why it is safe to ignore it?

Reply:
This is a pretty old piece from initial protocols commit https://github.com/python/mypy/commit/c55e48b5f1167cacefd15238ad80217f04b9891e 

I was not sure if this is safe to ignore or not. All code / tests I've tried showed that nothing has changed.

But, from personal experience finding bugs in constraints solver is pretty hard. 

We can ask @ilevkivskyi if he nows some examples / cases where this can backfire.

Reply:
This is a basic consistency check. One should never get here for something that can never be a protocol implementation. So I don't think this fixes the actual issue. Couple other comments:
* Variance in the place where it is used in the original repro should be totally irrelevant. Variance is only relevant for class definition type variables. It should be irrelevant in generic function/method type variables (including self-types). Variance is a property of a class, _not_ a property of a type variable. If the repro depends on variance we have a much bigger problem.
* I think the actual culprit is overload on self-type. This is a relatively new feature (much newer than protocols), and it requires some special handling in `bind_self()` (and/or somewhere around). I bet `find_member()` doesn't handle this logic correctly. So you should either fix that, or better refactor `find_member()` somehow to use the same logic.